executors/icu4j/74/executor-icu4j/src/main/java/org/unicode/conformance/testtype/datetimeformatter/DateTimeFormatterTester.java

```diff
@@ -156,10 +155,22 @@ public String getDateTimeFormatterResultString(DateTimeFormatterInputJson input)
 
     DateFormat dtf;
     if (input.skeleton != null) {
-      dtf = DateFormat.getInstanceForSkeleton(cal, input.skeleton, locale);
+      dtf = DateFormat.getInstanceForSkeleton(cal, input.skeleton, input.locale_with_calendar);
     } else {
-      dtf = DateFormat.getDateTimeInstance(cal, dateStyle, timeStyle, locale);
+      if (dateStyle >=0 && timeStyle >= 0) {
+        dtf = DateFormat.getDateTimeInstance(cal, dateStyle, timeStyle, input.locale_with_calendar);
+      } else
+        if (dateStyle >= 0) {
+          dtf = DateFormat.getDateInstance(cal, dateStyle,input.locale_with_calendar);
+        } else
+          if (timeStyle >= 0) {
+            dtf = DateFormat.getTimeInstance(cal, timeStyle, input.locale_with_calendar);
```

Comment:
Use consants here, and also simply use a `dateStyle != UNDEFINED_DATE_STYLE` rather than arithmetic. It's not wrong, but it's a very C++ idiom that is okay for indexes but not proper for enums IMHO.

Reply:
Yup!